lemmy_apub/src/objects/mod.rs

```diff
@@ -36,3 +39,22 @@ where
     Err(anyhow!("Cant convert object to tombstone if it wasnt deleted").into())
   }
 }
+
+pub(in crate::objects) fn check_object_domain<T, Kind>(
+  apub: &T,
+  expected_domain: Option<Url>,
+) -> Result<String, LemmyError>
+where
+  T: Base + AsBase<Kind>,
+{
+  let actor_id = if let Some(url) = expected_domain {
+    check_is_apub_id_valid(&url)?;
+    let domain = url.domain().context(location_info!())?;
+    apub.id(domain)?.context(location_info!())?
+  } else {
+    let actor_id = apub.id_unchecked().context(location_info!())?;
+    check_is_apub_id_valid(&actor_id)?;
+    actor_id
+  };
+  Ok(actor_id.to_string())
+}
```

Comment:
This is so incredibly confusing... I bet were triple verifying at times here. We're verifying on the activity domains, which does an object domain check, but then this appears to be another object domain check?

Reply:
The only change in this method is the first `check_is_apub_id_valid(&url)?` which I added (and the name change). 

And I would rather check things twice than missing a check. But yes its definitely confusing, and if you have any idea how to simplify things I'm all ears.

Reply:
mmk, I agree that multiple checks are harmless, but I'm gonna make an issue to go over and think of ways to simplify this checking.

Did you wanna do the shared_inbox as a commit after, and just keep this open?

Reply:
> Did you wanna do the shared_inbox as a commit after, and just keep this open?

Doesnt really make a difference for me, but maybe reviewing would be easier as separate PRs.